### PR TITLE
Add user table

### PR DESCRIPTION
**Pull request by cecoppinger**
So to test this you'll need to add a database using phpmyadmin.  The db name is "moore-market" with the same username and the password is "password".

### CODE REVIEW

File: MooreMarket/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        private MooreMarketContext _context;
+        public UserController(MooreMarketContext context)
+        {
+            _context = context;
+            _context.Database.EnsureCreated();
+            //some fake users
+            if (_context.Users.Count() == 0)
```
Comment (ryanis-online) [inline MooreMarket/Controllers/UserController.cs:21]: I'm fine with this being in here, but making a note of it, as we probably want to have a very specific database seeder for test data

File: MooreMarket/Data/Hash.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                                iterationCount: 10000,
+                                numBytesRequested: 256 / 8);
+ 
+            return Convert.ToBase64String(valueBytes);
+        }
+ 
+        public static bool Validate(string value, string salt, string hash)
+            => CreateHash(value, salt) == hash;
+
+        public static string MakeSalt()
```
Comment (ryanis-online) [inline MooreMarket/Data/Hash.cs:25]: Is this cryptographically secure?

File: MooreMarket/Startup.cs [hunk 3]
```diff
@@ -26,6 +22,8 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+            services.AddDbContext<MooreMarketContext>(options =>
+                options.UseMySQL("server=localhost;port=8889;database=moore-market;user=moore-market;password=password"));
```
Comment (ryanis-online) [inline MooreMarket/Startup.cs:26]: Can we store this in an `.env` environment file and load dynamically?

**Review by ryanis-online — CHANGES_REQUESTED** [review]
I made a couple of comments on specific commits, but overall:
1. Can you remove all generated files? (Anything that's a cache, or something built by the application)
2. Can you remove the `.vs` directory? I think it's best to leave these out of version control.

File: MooreMarket/Data/Hash.cs (hunk 2 above)
Comment (cecoppinger) [reply MooreMarket/Data/Hash.cs:25]: I'm not 100% certain about that I just kinda followed what I read from the msdn

File: MooreMarket/Startup.cs (hunk 3 above)
Comment (cecoppinger) [reply MooreMarket/Startup.cs:26]: I'm not exactly sure what you mean but I will try and figure that out

Comment (ryanis-online) [reply MooreMarket/Startup.cs:26]: This might help! https://github.com/bolorundurowb/dotenv.net

**Pull request closed without merging**